### ISSUE TITLE
Make it possible to switch locales at runtime

### ISSUE THREAD

**Issue by samreid** [phetsims/circuit-construction-kit-dc#199]
May be easier once https://github.com/phetsims/mean-share-and-balance/issues/99 is complete.

**Comment by samreid** [comment]
Also, on Thursday we discussed a `fillInDynamic` sort of method in StringUtils, which would help move this forward. I think @jonathanolson is assigned, but I couldn't see a paper trail.

**Comment by samreid** [comment]
I made a first pass at this and it's working well. Will need more attention once we have `fillInDynamic` and correct suffixes `StringProperty`.  Also, do I need to `addLinkedElement` on this?  I used `TReadOnlyProperty<string>` everywhere, but not sure if that's a match for studio autoselect.

**Comment by samreid** [comment]
Studio autoselect seems like it's working great.  All that's left here is waiting on 3 features:

* Strings renamed to e.g., currentStringProperty instead of currentProperty
* Support for fillInDynamic
* Ability to call `.value` or `.link` on StringProperty values inline.

On hold until there is progress on those features.

**Comment by jonathanolson** [comment]
I believe all of those items are done (*StringProperty, PatternStringProperty, and the .value/.link inline)

**Comment by samreid** [comment]
The remaining work for this issue is to convert `fillIn` to `PatternStringProperty`.  I started on that for the circuit element readouts (when values is checked).

<details>

```diff

Index: js/view/ValueNode.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/view/ValueNode.ts b/js/view/ValueNode.ts
--- a/js/view/ValueNode.ts	(revision 0233cf6cfa0edd8695fc9683441bb327ab172ee9)
+++ b/js/view/ValueNode.ts	(date 1662151035025)
@@ -31,6 +31,9 @@
 import CircuitElement from '../model/CircuitElement.js';
 import CircuitElementViewType from '../model/CircuitElementViewType.js';
 import Multilink from '../../../axon/js/Multilink.js';
+import PatternStringProperty from '../../../phetcommon/js/util/PatternStringProperty.js';
+import DerivedProperty, { DerivedPropertyOptions } from '../../../axon/js/DerivedProperty.js';
+import TReadOnlyProperty from '../../../axon/js/TReadOnlyProperty.js';
 
 const capacitanceFaradsSymbolStringProperty = circuitConstructionKitCommonStrings.capacitanceFaradsSymbolStringProperty;
 const fuseValueStringProperty = circuitConstructionKitCommonStrings.fuseValueStringProperty;
@@ -64,6 +67,12 @@
   font: FONT
 }, providedOptions ) );
 
+class ToFixedProperty extends DerivedProperty<string, number, never, never, never, never, never, never, never, never, never, never, never, never, never, never> {
+  public constructor( dependency: TReadOnlyProperty<number>, numberOfDecimalPlaces: number, options?: DerivedPropertyOptions<string> ) {
+    super( [ dependency ], ( value: number ) => Utils.toFixed( value, numberOfDecimalPlaces ), options );
+  }
+}
+
 const infinitySpan = '<span style="font-size: 26px; font-family: serif;"><b>∞</b></span>';
 
 export default class ValueNode extends Panel {
@@ -83,14 +92,14 @@
     let update: null | ( () => void ) = null;
     if ( circuitElement instanceof VoltageSource ) {
 
-      const voltageText = createText( tandem.createTandem( 'voltageText' ) );
-      const voltageListener = ( voltage: number ) => {
-
-        voltageText.text = StringUtils.fillIn( voltageUnitsStringProperty, {
-          voltage: Utils.toFixed( voltage, circuitElement.numberOfDecimalPlaces )
-        } );
-        update && update();
-      };
+      const voltageText = new Text( new PatternStringProperty( voltageUnitsStringProperty, {
+        voltage: new ToFixedProperty( circuitElement.voltageProperty, circuitElement.numberOfDecimalPlaces )
+      } ), {
+        tandem: tandem.createTandem( 'voltageText' ),
+        font: FONT
+      } );
+
+      const voltageListener = ( voltage: number ) => update && update();
       circuitElement.voltageProperty.link( voltageListener );
 
       // Battery readouts shows voltage and internal resistance if it is nonzero

```

</details>

However, I found that these are not even PhET-iO instrumented.  So it seems a question for the designers: which cases of `fillIn` should be phet-io instrumented and/or studio autoselectable?

* Readouts over the circuit elements
* Readouts in the edit container (at the bottom of the screen)
* Readouts for current and voltage on the meters.
* ChargeSpeedThrottleNode, saying what percent the sim has slowed.
